### PR TITLE
Add GetLatestApprovers to Platform interface

### PR DESCRIPTION
**Pull request by gjonathanhong**

### CODE REVIEW

File: pkg/platform/github.go [hunk 1]
```diff
@@ -150,6 +154,96 @@ func (g *GitHub) AssignReviewers(ctx context.Context, input *AssignReviewersInpu
...
+			Nodes []team
+		} `graphql:"teams(first: 100)"`
+	} `graphql:"organization(login: $owner)"`
+}
+
+// GetLatestApprovers retrieves the users whose latest review for a pull request
+// is an approval. It also returns the teams and subteams that the user
+// approvers are members of to indicate approval on behalf of those teams. Note,
+// if a reviewer leaves a comment OR requests changes after a previous approval,
+// then the reviewer will not be included in the "latest" approvers list.
```
Comment (verbanicm) [inline pkg/platform/github.go:195]: Is this the behavior we want? I know we've touched on this, but I would think any approval is an approval.

File: pkg/platform/github.go [hunk 2]
```diff
@@ -41,8 +42,9 @@ var (
 
 // GitHub implements the Platform interface.
 type GitHub struct {
-	cfg    *gh.Config
-	client *github.Client
+	cfg           *gh.Config
+	client        *github.Client
+	graphQLClient *githubv4.Client
```
Comment (verbanicm) [inline pkg/platform/github.go:47]: nit: I think the "appropriate" naming would be graphqlClient

File: pkg/platform/github.go (hunk 1 above)
Comment (gjonathanhong) [reply pkg/platform/github.go]: I just tested the behavior. It looks like a comment will not change the status of the review, so a prior approval will still be in effect. If the approval is dismissed or followed by a requested changes, then the approval won't be counted.

That does seem like the behavior we want.

Comment (verbanicm) [reply pkg/platform/github.go]: +1, thanks!

File: pkg/platform/github.go [hunk 3]
```diff
@@ -150,6 +154,97 @@ func (g *GitHub) AssignReviewers(ctx context.Context, input *AssignReviewersInpu
...
+	hasApproved := make(map[string]struct{}, len(approversQuery.Repository.PullRequest.LatestReviews.Nodes))
+	for _, review := range approversQuery.Repository.PullRequest.LatestReviews.Nodes {
+		if review.State == "APPROVED" {
+			result.Users = append(result.Users, review.Author.Login)
+			hasApproved[review.Author.Login] = struct{}{}
+		}
+	}
+
+	var teamQuery organizationTeamsAndMembershipsQuery
+	if err := g.graphqlClient.Query(ctx, &teamQuery, map[string]any{
```
Comment (verbanicm) [inline pkg/platform/github.go:225]: followup: should we paginate this? I would guess some orgs might have over 100 teams, users, etc

Example: https://github.com/abcxyz/github-metrics-aggregator/blob/main/pkg/review/commit_review_status.go

**Review by verbanicm — APPROVED** [review]

**Pull request merged**